### PR TITLE
Chlodochar pr removedist

### PR DESCRIPTION
**Pull request by chuyskywalker**
develp merge + dockerfile updates for a fully working, ready to distribute container

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -4,16 +4,17 @@ module.exports = function(grunt) {
...
+          'static/css/main.css' : 'static/sass/main.scss',
+          'static/css/mobile.css': 'static/sass/mobile.scss'
+        }
+      }
+    },
     jshint: {
       files: ['Gruntfile.js', 'js/*.js', '!js/vendor/**/*.js'],
       options: {
-		    reporter: require('jshint-stylish'),
+        reporter: require('jshint-stylish'),
```
Comment (Chlodochar) [inline Gruntfile.js:17]: **[eslint]**
- Error - 'require' is not defined. (no-undef)

File: static/js/mobile.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
+var loadingMessage = document.getElementById('msg');
```
Comment (Chlodochar) [inline static/js/mobile.js:1]: **[eslint]**
- Error - 'document' is not defined. (no-undef)

File: static/js/mobile.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+var useLoc = document.getElementById('use-loc');
```
Comment (Chlodochar) [inline static/js/mobile.js:3]: **[eslint]**
- Error - 'document' is not defined. (no-undef)

File: static/js/mobile.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+useLoc.checked = localStorage.useLoc === 'true';
```
Comment (Chlodochar) [inline static/js/mobile.js:4]: **[eslint]**
- Error - 'localStorage' is not defined. (no-undef)

File: static/js/mobile.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+useLoc.onchange = function() {
+	localStorage.useLoc = useLoc.checked;
```
Comment (Chlodochar) [inline static/js/mobile.js:6]: **[eslint]**
- Error - 'localStorage' is not defined. (no-undef)

File: static/js/mobile.js [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+};
+
+var tm;
+var useTimer = document.getElementById('use-timer');
```
Comment (Chlodochar) [inline static/js/mobile.js:10]: **[eslint]**
- Error - 'document' is not defined. (no-undef)

File: static/js/mobile.js [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+useTimer.checked = localStorage.useTimer !== 'false';
```
Comment (Chlodochar) [inline static/js/mobile.js:11]: **[eslint]**
- Error - 'localStorage' is not defined. (no-undef)

File: static/js/mobile.js [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+useTimer.onchange = function() {
+	localStorage.useTimer = useTimer.checked;
```
Comment (Chlodochar) [inline static/js/mobile.js:13]: **[eslint]**
- Error - 'localStorage' is not defined. (no-undef)

File: static/js/mobile.js [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+	if (localStorage.useTimer === 'false') {
```
Comment (Chlodochar) [inline static/js/mobile.js:14]: **[eslint]**
- Error - 'localStorage' is not defined. (no-undef)

File: static/js/mobile.js [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+		clearTimeout(tm);
```
Comment (Chlodochar) [inline static/js/mobile.js:15]: **[eslint]**
- Error - 'clearTimeout' is not defined. (no-undef)

File: static/js/mobile.js [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+	}
+	else {
+		clearTimeout(tm);
```
Comment (Chlodochar) [inline static/js/mobile.js:18]: **[eslint]**
- Error - 'clearTimeout' is not defined. (no-undef)

File: static/js/mobile.js [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+		updateTimes();
+	}
+};
+
+document.querySelector('#nav button').onclick = function() {
```
Comment (Chlodochar) [inline static/js/mobile.js:23]: **[eslint]**
- Error - 'document' is not defined. (no-undef)

File: static/js/mobile.js [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+	if (localStorage.useLoc !== 'true') {
```
Comment (Chlodochar) [inline static/js/mobile.js:24]: **[eslint]**
- Error - 'localStorage' is not defined. (no-undef)

File: pogom/app.py [hunk 14]
```diff
@@ -126,6 +126,7 @@ def list_pokemon(self):
                 'time_to_disappear': '%d min %d sec' % (divmod((
                     pokemon['disappear_time']-datetime.utcnow()).seconds, 60)),
                 'disappear_time': pokemon['disappear_time'],
+                'disappear_sec': (pokemon['disappear_time']-datetime.utcnow()).seconds,
```
Comment (Chlodochar) [inline pogom/app.py:129]: **[flake8]**
- [E501] line too long (87 > 79 characters)

File: static/js/mobile.js [hunk 15]
```diff
@@ -0,0 +1,75 @@
...
+		return (location.href="/mobile");
```
Comment (Chlodochar) [inline static/js/mobile.js:25]: **[eslint]**
- Error - 'location' is not defined. (no-undef)

File: pogom/search.py [hunk 16]
```diff
@@ -61,47 +52,61 @@ def send_map_request(api, position):
         log.warning("Uncaught exception when downloading map " + str(e))
         return False
 
+def get_new_coords(init_loc, distance, bearing):
```
Comment (Chlodochar) [inline pogom/search.py:55]: **[flake8]**
- [E302] expected 2 blank lines, found 1

File: static/js/mobile.js [hunk 17]
```diff
@@ -0,0 +1,75 @@
...
+	}
+	else if ("geolocation" in navigator) {
```
Comment (Chlodochar) [inline static/js/mobile.js:27]: **[eslint]**
- Error - 'navigator' is not defined. (no-undef)

File: pogom/search.py [hunk 18]
```diff
@@ -61,47 +52,61 @@ def send_map_request(api, position):
...
+    """ Given an initial lat/lng, a distance(in kms), and a bearing (degrees),
+    this will calculate the resulting lat/lng coordinates.
+    """
```
Comment (Chlodochar) [inline pogom/search.py:58]: **[flake8]**
- [W291] trailing whitespace

File: static/js/mobile.js [hunk 19]
```diff
@@ -0,0 +1,75 @@
...
+
+		// Getting the GPS position can be very slow on some devices
+		loadingMessage.innerText='Getting location...';
+		navigator.geolocation.getCurrentPosition(function(p) {
```
Comment (Chlodochar) [inline static/js/mobile.js:31]: **[eslint]**
- Error - 'navigator' is not defined. (no-undef)

File: pogom/search.py [hunk 20]
```diff
@@ -61,47 +52,61 @@ def send_map_request(api, position):
         log.warning("Uncaught exception when downloading map " + str(e))
         return False
 
+def get_new_coords(init_loc, distance, bearing):
+    """ Given an initial lat/lng, a distance(in kms), and a bearing (degrees),
+    this will calculate the resulting lat/lng coordinates.
+    """ 
+    R = 6378.1 #km radius of the earth
```
Comment (Chlodochar) [inline pogom/search.py:59]: **[flake8]**
- [E261] at least two spaces before inline comment
- [E262] inline comment should start with '# '

File: static/js/mobile.js [hunk 21]
```diff
@@ -0,0 +1,75 @@
...
+			loadingMessage.innerText='Reloading...';
+			location.href='/mobile?lat='+p.coords.latitude+'&lon='+p.coords.longitude;
```
Comment (Chlodochar) [inline static/js/mobile.js:33]: **[eslint]**
- Error - 'location' is not defined. (no-undef)

File: pogom/search.py [hunk 22]
```diff
@@ -61,47 +52,61 @@ def send_map_request(api, position):
...
-    lng_location = initial_location[1]
-
-    yield (initial_location[0], initial_location[1], 0)  # Middle circle
-
-    while ring < num_steps:
-        # Move the location diagonally to top left spot, then start the circle which will end up back here for the next ring
-        # Move Lat north first
-        lat_location += lat_gap_degrees
-        lng_location -= calculate_lng_degrees(lat_location)
+    init_coords = [math.radians(init_loc[0]), math.radians(init_loc[1])] # convert lat/lng to radians
```
Comment (Chlodochar) [inline pogom/search.py:62]: **[flake8]**
- [E261] at least two spaces before inline comment
- [E501] line too long (101 > 79 characters)

File: static/js/mobile.js [hunk 23]
```diff
@@ -0,0 +1,75 @@
...
+
+		}, function(err) {
+			loadingMessage.innerText='';
+			alert('Failed to get location: ' + err.message);
```
Comment (Chlodochar) [inline static/js/mobile.js:37]: **[eslint]**
- Error - 'alert' is not defined. (no-undef)

File: pogom/search.py [hunk 24]
```diff
@@ -61,47 +52,61 @@ def send_map_request(api, position):
...
 
-        for direction in range(6):
-            for i in range(ring):
-                if direction == 0:  # Right
-                    lng_location += calculate_lng_degrees(lat_location) * 2
-
-                if direction == 1:  # Right Down
-                    lat_location -= lat_gap_degrees
-                    lng_location += calculate_lng_degrees(lat_location)
+    new_lat = math.asin( math.sin(init_coords[0])*math.cos(distance/R) +
```
Comment (Chlodochar) [inline pogom/search.py:64]: **[flake8]**
- [E201] whitespace after '('

File: static/js/mobile.js [hunk 25]
```diff
@@ -0,0 +1,75 @@
...
+
+		}, {
+			enableHighAccuracy: true,
+			timeout: 5000,
+			maximumAge: 5000
+		});
+	}
+	else {
+		alert('Your device does not support web geolocation');
```
Comment (Chlodochar) [inline static/js/mobile.js:46]: **[eslint]**
- Error - 'alert' is not defined. (no-undef)

File: pogom/search.py [hunk 26]
```diff
@@ -61,47 +52,61 @@ def send_map_request(api, position):
...
+        math.cos(init_coords[0])*math.sin(distance/R)*math.cos(bearing))
```
Comment (Chlodochar) [inline pogom/search.py:65]: **[flake8]**
- [E128] continuation line under-indented for visual indent

File: pogom/search.py [hunk 27]
```diff
@@ -61,47 +52,61 @@ def send_map_request(api, position):
...
 
-                if direction == 2:  # Left Down
-                    lat_location -= lat_gap_degrees
-                    lng_location -= calculate_lng_degrees(lat_location)
+    new_lon = init_coords[1] + math.atan2(math.sin(bearing)*math.sin(distance/R)*math.cos(init_coords[0]),
```
Comment (Chlodochar) [inline pogom/search.py:67]: **[flake8]**
- [E501] line too long (106 > 79 characters)

File: static/js/mobile.js [hunk 28]
```diff
@@ -0,0 +1,75 @@
...
+	}
+};
+
+function updateTimes() {
+	// server tells us how many seconds are left we note the
+	// pageload time and count down from there.
+	// Yes, this could be a smidge innaccurate, but not by
+	// more than 1 second or so which doesn't matter.
+	// And now we don't have to deal with timestamps and dates!
+	document.querySelectorAll('div.remain').forEach(function(element){
```
Comment (Chlodochar) [inline static/js/mobile.js:56]: **[eslint]**
- Error - 'document' is not defined. (no-undef)

File: pogom/search.py [hunk 29]
```diff
@@ -61,47 +52,61 @@ def send_map_request(api, position):
...
+        math.cos(distance/R)-math.sin(init_coords[0])*math.sin(new_lat))
```
Comment (Chlodochar) [inline pogom/search.py:68]: **[flake8]**
- [E128] continuation line under-indented for visual indent

File: static/js/mobile.js [hunk 30]
```diff
@@ -0,0 +1,75 @@
...
+		var now = new Date().getTime();
+		var secondsPassed = Math.floor((now - page_loaded)/1000);
```
Comment (Chlodochar) [inline static/js/mobile.js:58]: **[eslint]**
- Error - 'page_loaded' is not defined. (no-undef)

File: pogom/search.py [hunk 31]
```diff
@@ -61,47 +52,61 @@ def send_map_request(api, position):
...
 
-                if direction == 3:  # Left
-                    lng_location -= calculate_lng_degrees(lat_location) * 2
+    return [math.degrees(new_lat), math.degrees(new_lon)]
 
-                if direction == 4:  # Left Up
-                    lat_location += lat_gap_degrees
-                    lng_location -= calculate_lng_degrees(lat_location)
+def generate_location_steps(initial_loc, step_count):
```
Comment (Chlodochar) [inline pogom/search.py:72]: **[flake8]**
- [E302] expected 2 blank lines, found 1

File: static/js/mobile.js [hunk 32]
```diff
@@ -0,0 +1,75 @@
...
+		var alivefor =  element.getAttribute('disappear');
+		var remain = alivefor - secondsPassed;
+		var min = Math.floor(remain / 60);
+		var sec = remain % 60;
+		element.innerText = (remain > 0) ? min + ' min ' + sec + ' sec' : '(expired)';
+	});
+	tm = setTimeout(updateTimes, 1000);
```
Comment (Chlodochar) [inline static/js/mobile.js:65]: **[eslint]**
- Error - 'setTimeout' is not defined. (no-undef)

File: pogom/search.py [hunk 33]
```diff
@@ -61,47 +52,61 @@ def send_map_request(api, position):
...
+    #Bearing (degrees)
```
Comment (Chlodochar) [inline pogom/search.py:73]: **[flake8]**
- [E265] block comment should start with '# '

File: static/js/mobile.js [hunk 34]
```diff
@@ -0,0 +1,75 @@
...
+};
```
Comment (Chlodochar) [inline static/js/mobile.js:66]: **[eslint]**
- Error - Unnecessary semicolon. (no-extra-semi)

File: pogom/search.py [hunk 35]
```diff
@@ -61,47 +52,61 @@ def send_map_request(api, position):
...
+    NORTH = 0
+    EAST = 90
+    SOUTH = 180
+    WEST = 270
 
-                if direction == 5:  # Right Up
-                    lat_location += lat_gap_degrees
-                    lng_location += calculate_lng_degrees(lat_location)
+    pulse_radius = 0.1                  # km - radius of players heartbeat is 100m
```
Comment (Chlodochar) [inline pogom/search.py:79]: **[flake8]**
- [E501] line too long (82 > 79 characters)

File: static/js/mobile.js [hunk 36]
```diff
@@ -0,0 +1,75 @@
...
+if (localStorage.useTimer !== 'false') {
```
Comment (Chlodochar) [inline static/js/mobile.js:67]: **[eslint]**
- Error - 'localStorage' is not defined. (no-undef)

File: static/js/mobile.js [hunk 37]
```diff
@@ -0,0 +1,75 @@
...
+	updateTimes();
+}
+
+document.querySelectorAll("li").forEach(function(listItem) {
```
Comment (Chlodochar) [inline static/js/mobile.js:71]: **[eslint]**
- Error - 'document' is not defined. (no-undef)

File: static/js/mobile.js [hunk 38]
```diff
@@ -0,0 +1,75 @@
...
+	listItem.onclick = function() {
+		window.document.location = this.getAttribute('href');
```
Comment (Chlodochar) [inline static/js/mobile.js:73]: **[eslint]**
- Error - 'window' is not defined. (no-undef)

**Comment by Chlodochar** [comment]
Click here if you want to create eslint's auto-correct Pull Request 
[Auto Correct by SideCI](https://sideci.com/github_repositories/63815592/issues/pull_request/1/diff/17b85a4b24fb795167c91d57e961f803b3657099?open_modal=1#eslint)

See the documentation if you want to get more information about auto-correct.
[Document](https://sideci.com/ja/docs/autofix-eslint)

File: pogom/search.py [hunk 39]
```diff
@@ -61,47 +52,61 @@ def send_map_request(api, position):
...
+    xdist = math.sqrt(3)*pulse_radius   # dist between column centers
+    ydist = 3*(pulse_radius/2)          # dist between row centers
 
-                yield (lat_location, lng_location, 0)  # Middle circle
+    yield (initial_loc[0], initial_loc[1], 0) #insert initial location
```
Comment (Chlodochar) [inline pogom/search.py:83]: **[flake8]**
- [E261] at least two spaces before inline comment
- [E262] inline comment should start with '# '

File: pogom/search.py [hunk 40]
```diff
@@ -61,47 +52,61 @@ def send_map_request(api, position):
...
 
+    ring = 1
```
Comment (Chlodochar) [inline pogom/search.py:85]: **[flake8]**
- [W291] trailing whitespace

File: pogom/search.py [hunk 41]
```diff
@@ -61,47 +52,61 @@ def send_map_request(api, position):
...
+    xdist = math.sqrt(3)*pulse_radius   # dist between column centers
+    ydist = 3*(pulse_radius/2)          # dist between row centers
 
-                yield (lat_location, lng_location, 0)  # Middle circle
+    yield (initial_loc[0], initial_loc[1], 0) #insert initial location
 
+    ring = 1            
+    loc = initial_loc
+    while ring < step_count:
+        #Set loc to start at top left
```
Comment (Chlodochar) [inline pogom/search.py:88]: **[flake8]**
- [E265] block comment should start with '# '

File: pogom/search.py [hunk 42]
```diff
@@ -61,47 +52,61 @@ def send_map_request(api, position):
...
+        loc = get_new_coords(loc, ydist, NORTH)
+        loc = get_new_coords(loc, xdist/2, WEST)
+        for direction in range(6):
+            for i in range(ring):
+                if direction == 0: # RIGHT
```
Comment (Chlodochar) [inline pogom/search.py:93]: **[flake8]**
- [E261] at least two spaces before inline comment

File: pogom/search.py [hunk 43]
```diff
@@ -61,47 +52,61 @@ def send_map_request(api, position):
...
+                    loc = get_new_coords(loc, xdist, EAST)
+                if direction == 1: # DOWN + RIGHT
```
Comment (Chlodochar) [inline pogom/search.py:95]: **[flake8]**
- [E261] at least two spaces before inline comment

File: pogom/search.py [hunk 44]
```diff
@@ -61,47 +52,61 @@ def send_map_request(api, position):
...
+                    loc = get_new_coords(loc, ydist, SOUTH)
+                    loc = get_new_coords(loc, xdist/2, EAST)
+                if direction == 2: # DOWN + LEFT
```
Comment (Chlodochar) [inline pogom/search.py:98]: **[flake8]**
- [E261] at least two spaces before inline comment

File: pogom/search.py [hunk 45]
```diff
@@ -61,47 +52,61 @@ def send_map_request(api, position):
...
+                    loc = get_new_coords(loc, ydist, SOUTH)
+                    loc = get_new_coords(loc, xdist/2, WEST)
+                if direction == 3: # LEFT
```
Comment (Chlodochar) [inline pogom/search.py:101]: **[flake8]**
- [E261] at least two spaces before inline comment

File: pogom/search.py [hunk 46]
```diff
@@ -61,47 +52,61 @@ def send_map_request(api, position):
...
+                    loc = get_new_coords(loc, xdist, WEST)
+                if direction == 4: # UP + LEFT
```
Comment (Chlodochar) [inline pogom/search.py:103]: **[flake8]**
- [E261] at least two spaces before inline comment

File: pogom/search.py [hunk 47]
```diff
@@ -61,47 +52,61 @@ def send_map_request(api, position):
...
+                    loc = get_new_coords(loc, ydist, NORTH)
+                    loc = get_new_coords(loc, xdist/2, WEST)
+                if direction == 5: # UP + RIGHT
```
Comment (Chlodochar) [inline pogom/search.py:106]: **[flake8]**
- [E261] at least two spaces before inline comment

**Comment by chuyskywalker** [comment]
"39 messages, what the shit!?!"

"Oh, stupid auto lint responses..."

**Pull request closed without merging**